can_tools/scrapers/official/base.py

```diff
@@ -1222,20 +1223,55 @@ def get_dataset(self, body, url) -> str:
         return rawJson
 
 
-class ETagCacheMixin:
+class CacheMixin(ABC):
```

Comment:
Create an abstract class that will be used by different caching implementations (for now just the old Etag caching, and the system used for the CDC scraper)